docs/src/test-parameterize-js.md

```diff
@@ -9,15 +9,21 @@ You can either parameterize tests on a test level or on a project level.
 ## Parameterized Tests
 
 ```js title="example.spec.ts"
-const people = ['Alice', 'Bob'];
-for (const name of people) {
-  test(`testing with ${name}`, async () => {
-    // ...
-  });
+[
+  { name: 'Alice', expected: 'Hello, Alice!' },
+  { name: 'Bob', expected: 'Hello, Bob!' },
+  { name: 'Charlie', expected: 'Hello, Charlie!' },
+].forEach(({ name, expected }) => {
   // You can also do it with test.describe() or with multiple tests as long the test name is unique.
-}
+  test(`testing with ${name}`, async ({ page }) => {
+    await page.goto(`/greet?name=${name}`);
+    await expect(page.getByRole('heading', { name })).toBeVisible();
+  });
+});
 ```
 
+When looping over `before{Each,All}` or `after{Each,All}` hooks, they will be executed for every iteration, so in most cases, you should loop only over tests.
```

Comment:
This is not really clear. An example would be nice. I'd do a subsection `### Before and after hooks` right here.